services/src/tasks/mod.rs

```diff
@@ -0,0 +1,141 @@
+mod error;
+mod in_memory;
+
+use crate::error::Result;
+pub use error::TaskError;
+use geoengine_datatypes::{error::ErrorSource, identifier};
+pub use in_memory::{InMemoryTaskDb, InMemoryTaskDbContext};
+use serde::{Deserialize, Serialize, Serializer};
+use std::{fmt, sync::Arc};
+
+/// A database that allows registering and retrieving tasks.
+#[async_trait::async_trait]
+pub trait TaskDb<C: TaskContext>: Send + Sync {
+    async fn register(&self, task: Box<dyn Task<C>>) -> Result<TaskId, TaskError>;
+
+    async fn status(&self, task_id: TaskId) -> Result<TaskStatus, TaskError>;
+
+    async fn list(&self, options: TaskListOptions) -> Result<Vec<TaskStatusWithId>, TaskError>;
```

Comment:
should be `Validated<TaskListOptions>` for uniformity

Reply:
[905662a](https://github.com/geo-engine/geoengine/pull/555/commits/905662ae2828c81f18a9cde0de83cd8257717d4f)